### PR TITLE
AnimatedTexture: Fix crash when loaded from a thread

### PR DESCRIPTION
**Pull request by RandomShaper**
Signals (like most, or all?, of the `Object` API) is not thread-safe. `AnimatedTexture` tries to connect to `RenderingServer`'s singal `frame_pre_draw` in the constructor. The issue is that such constructor may be running on any thread.

An alternative solution would be to make all the signal API in `Object` virtual so that `RenderingServer` can wrap all the functions with the `server_wrap_mt_common.h` utilities. However, that would harm performance in every other case.

In conclusion, it's OK to consider the specification for writing resource classes to include it should only deal with thread-safe APIs.

### CODE REVIEW

File: scene/resources/animated_texture.cpp [hunk 1]
```diff
@@ -270,13 +270,18 @@ void AnimatedTexture::_bind_methods() {
...
+
 AnimatedTexture::AnimatedTexture() {
 	//proxy = RS::get_singleton()->texture_create();
 	proxy_ph = RS::get_singleton()->texture_2d_placeholder_create();
 	proxy = RS::get_singleton()->texture_proxy_create(proxy_ph);
 
 	RenderingServer::get_singleton()->texture_set_force_redraw_if_visible(proxy, true);
-	RenderingServer::get_singleton()->connect("frame_pre_draw", callable_mp(this, &AnimatedTexture::_update_proxy));
+
+	MessageQueue::get_main_singleton()->push_callable(callable_mp(this, &AnimatedTexture::_finish_non_thread_safe_setup));
```
Comment (akien-mga) [inline scene/resources/animated_texture.cpp:284]: Wouldn't using the `CONNECT_DEFERRED` flag achieve something similar without having to introduce a new method?

Comment (RandomShaper) [reply scene/resources/animated_texture.cpp:284]: Good question, but this is another case of the difference between the main call queue and the thread override call queue is relevant. `CONNECT_DEFERRED` would use `MessageQueue::get_singleton()`, which during threaded loading wouldn't be the good old default `MessageQueue`, but the override the `ResourceLoader` has set for the loading thread. We need to bypass it.

Maybe we could consider making that a bit more transparent. However, as I see things, we are in a _de facto_ stage of colllection of information about the needs of resources dealing with a multi-threaded world and we're not ready yet to come up with solid API designs that are better than the manual approaches.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
